app/environments/brassbirmingham/brassbirmingham/envs/classes/board.py

```diff
@@ -423,6 +480,22 @@ def getAvailableCoalBuildingsTradePosts(
                     l.append(tradePost)
```

Comment:
Not related to this MR, but we don't need the `coalMarketRemaining > 0` check. Even if the coal market is 0, there's still an infinite supply of coal from trade posts. Just expensive